src/event-engine/states/handshake_failure.ts

```diff
@@ -1,9 +1,10 @@
 import { State } from '../core/state';
```

Comment:
`HANDSHAKE_FAILED` is missing handling for `SUBSCRIPTION_CHANGED`, `SUBSCRIPTION_RESTORED` and `UNSUBSCRIBE_ALL`.

Reply:
Added all 3 ✔️ 

https://github.com/pubnub/javascript/blob/c35559fef8bfb49b63d334fd42fb3733f36c75bd/src/event-engine/states/handshake_failed.ts#L18